externals/kyuubi-flink-sql-engine/src/main/java/org/apache/kyuubi/engine/flink/result/ResultSet.java

```diff
@@ -49,6 +53,7 @@ public class ResultSet {
   // list of boolean in streaming mode,
   // true if the corresponding row is an append row, false if its a retract row
   private final List<Boolean> changeFlags;
+  public static boolean versionAtLeast15 = FlinkEngineUtils.isFlinkVersionAtLeast("1.15");
```

Comment:
Pullout everything to the field is not a good practice. If logic/variables are not complex and not used multi times, please keep them inline and local scope.

Reply:
Then restore the original and put it in the fromTableResult method?

Reply:
The following style should be clear enough.
```
static import org.apache.kyuubi.engine.flink.FlinkEngineUtils.isFlinkVersionAtLeast

if (isFlinkVersionAtLeast("1.15")) {
  ...
} else {
  ...
}
```